### PR TITLE
MULE-11131 Improve implementation of filtering/request-response proce…

### PR DESCRIPTION
**Pull request by dfeist**
…ssors to avoid concatMaps

### CODE REVIEW

File: core/src/main/java/org/mule/runtime/core/processor/strategy/ReactorProcessingStrategyFactory.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+ *
+ * This processing strategy is not suitable for transactional flows and will fail if used with an active transaction.
+ *
+ * @since 4.0
+ */
+public class ReactorProcessingStrategyFactory implements ProcessingStrategyFactory {
+
+  @Override
+  public ProcessingStrategy create(MuleContext muleContext) {
+    return new ReactorProcessingStrategy(() -> muleContext.getSchedulerService().customScheduler("event-loop", 1),
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/processor/strategy/ReactorProcessingStrategyFactory.java]: could this be a throttled cpu-light now?

File: transports/core-transports-tests/src/test/java/org/mule/compatibility/core/endpoint/outbound/OutboundResponsePropertiesMessageProcessorTestCase.java [hunk 2]
```diff
@@ -1,60 +0,0 @@
...
-import org.mule.compatibility.core.endpoint.AbstractEndpointBuilder;
-import org.mule.compatibility.core.processor.AbstractMessageProcessorTestCase;
-import org.mule.runtime.core.api.Event;
-import org.mule.runtime.core.api.message.InternalMessage;
-import org.mule.runtime.core.api.processor.InterceptingMessageProcessor;
-import org.mule.runtime.core.session.DefaultMuleSession;
-
-import org.junit.Test;
-
-public class OutboundResponsePropertiesMessageProcessorTestCase extends AbstractMessageProcessorTestCase {
```
Comment (elrodro83) [inline transports/core-transports-tests/src/test/java/org/mule/compatibility/core/endpoint/outbound/OutboundResponsePropertiesMessageProcessorTestCase.java]: why remove this?

File: transports/core/src/main/java/org/mule/compatibility/core/endpoint/DefaultEndpointMessageProcessorChainFactory.java [hunk 3]
```diff
@@ -76,7 +75,6 @@
     list.add(new OutboundSessionHandlerMessageProcessor(connector.getSessionHandler(), endpoint.getMuleContext()));
     list.add(new OutboundEndpointPropertyMessageProcessor(endpoint));
     list.add(new OutboundRootMessageIdPropertyMessageProcessor());
-    list.add(new OutboundResponsePropertiesMessageProcessor(endpoint));
```
Comment (elrodro83) [inline transports/core/src/main/java/org/mule/compatibility/core/endpoint/DefaultEndpointMessageProcessorChainFactory.java]: why remove this?

File: transports/core/src/main/java/org/mule/compatibility/core/endpoint/outbound/OutboundResponsePropertiesMessageProcessor.java [hunk 4]
```diff
@@ -1,51 +0,0 @@
...
-
-import java.io.Serializable;
-
-/**
- * Propagates properties from request message to response message as defined by {@link OutboundEndpoint#getResponseProperties()}.
- * <p>
- * //TODO This can became a standard MessageProcessor in the response chain if/when event has a (immutable) reference to request
- * message.
- */
-public class OutboundResponsePropertiesMessageProcessor extends AbstractRequestResponseMessageProcessor {
```
Comment (elrodro83) [inline transports/core/src/main/java/org/mule/compatibility/core/endpoint/outbound/OutboundResponsePropertiesMessageProcessor.java]: why remove this?

File: transports/core/src/main/java/org/mule/compatibility/core/transport/AbstractMessageDispatcher.java [hunk 5]
```diff
@@ -83,7 +86,16 @@ private Event createResponseEvent(InternalMessage resultMessage, Event requestEv
                                                                                                resultMessage,
                                                                                                endpoint.getMuleContext());
       requestEvent.getSession().merge(storedSession);
-      Event resultEvent = Event.builder(requestEvent).message(resultMessage).build();
+
+      InternalMessage.Builder messageBuilder = InternalMessage.builder(resultMessage);
```
Comment (elrodro83) [inline transports/core/src/main/java/org/mule/compatibility/core/transport/AbstractMessageDispatcher.java]: why is this needed?

File: core/src/main/java/org/mule/runtime/core/processor/strategy/ReactorProcessingStrategyFactory.java (hunk 1 above)
Comment (dfeist) [inline core/src/main/java/org/mule/runtime/core/processor/strategy/ReactorProcessingStrategyFactory.java]: No, but i should add a TODO.

File: transports/core-transports-tests/src/test/java/org/mule/compatibility/core/endpoint/outbound/OutboundResponsePropertiesMessageProcessorTestCase.java (hunk 2 above)
Comment (dfeist) [inline transports/core-transports-tests/src/test/java/org/mule/compatibility/core/endpoint/outbound/OutboundResponsePropertiesMessageProcessorTestCase.java]: Because it depends on behaviour of AbstractRequestResponseMP which no longer exists.

File: transports/core/src/main/java/org/mule/compatibility/core/transport/AbstractMessageDispatcher.java (hunk 5 above)
Comment (dfeist) [inline transports/core/src/main/java/org/mule/compatibility/core/transport/AbstractMessageDispatcher.java]: Because the other MP was removed :-)
